In [35]:
import numpy as np
import pandas as pd
from sklearn.metrics import mean_absolute_error

In [31]:
daydf = pd.read_csv("../data/raw/day.csv")

In [32]:
daydf["baseline_persistence"] = daydf["cnt"].shift(1)
daydf["baseline_seasonal"] = daydf["cnt"].shift(7)

split = int(len(daydf) * 0.8) # train 80%, test 20%
train = daydf.iloc[:split]
test = daydf.iloc[split:]

In [33]:
# baseline 1: y chang ngày hôm qua
mae1 = mean_absolute_error(test["cnt"], test["baseline_persistence"])
print(f"Persistence baseline MAE: {mae1:.2f}")

Persistence baseline MAE: 878.39


In [34]:
# baseline 2: giống ngày này tuần trước
mae2 = mean_absolute_error(test['cnt'], test['baseline_seasonal'])
print(f"Seasonal baseline MAE: {mae2:.2f}")

Seasonal baseline MAE: 1194.73


# summaryy
- Baseline Persistence đạt MAE thấp hơn Baseline Seasonal (878 so với 1194), do đó được chọn làm baseline chính thức cho các mô hình học máy ở các bước tiếp theo.